src/utils/prototype-for.js

```diff
@@ -0,0 +1,146 @@
+import MicroState from '../microstate';
+
+const { getOwnPropertyNames, getOwnPropertyDescriptor, defineProperty } = Object;
+
+export default function prototypeFor(prototype, attrs) {
```

Comment:
A perhaps different way of saying it is that I think the key thing that we're fighting is that I think we want to generate a prototype per instance (which has the class prototype as _it's_ prototype, but with it's own overrides to represent the properties it uniquely holds) as opposed to a single class prototype that handles all cases.